build/build.go

```diff
@@ -1523,3 +1540,47 @@ func ReadSourcePolicy() (*spb.Policy, error) {
 
 	return &pol, nil
 }
+
+// backendAttribute is a utility to retrieve the backend attribute from a resolvedNode.
+func backendAttribute(dp *resolvedNode) attribute.KeyValue {
+	driverName := dp.Node().Driver.Factory().Name()
+	return attribute.String("backend", driverName)
+}
+
+// buildIDAttribute is a utility to retrieve the build id attribute from the solve options.
+// This value should be consistent between builds.
+func buildIDAttribute(so *client.SolveOpt) attribute.KeyValue {
+	vcs := so.FrontendAttrs["vcs:source"]
+	target := so.FrontendAttrs["target"]
+	context := so.FrontendAttrs["context"]
+	filename := so.FrontendAttrs["filename"]
+
+	buildID := ""
+	if vcs != "" || target != "" || context != "" || filename != "" {
+		h := xxh3.New()
```

Comment:
We use sha256 everywhere. This is not a place where performance considerations would require new dependency.

Reply:
Updated to use sha256.